manager/controllers/app/m4dapplication_controller.go

```diff
@@ -222,34 +226,24 @@ func (r *M4DApplicationReconciler) reconcile(applicationContext *app.M4DApplicat
 	if hasError(applicationContext) {
 		return ctrl.Result{}, nil
 	}
-	r.Log.V(0).Info("Creating Blueprint")
-	// first, a namespace should be created
-	if applicationContext.Status.BlueprintNamespace == "" {
-		if err := r.CreateNamespace(applicationContext); err != nil {
-			return ctrl.Result{}, err
-		}
-	}
 	// unite several instances of a read/write module
 	newInstances := r.RefineInstances(instances)
+	r.Log.V(0).Info("Creating blueprint(s)")
+
 	blueprintSpec := r.GenerateBlueprint(newInstances, applicationContext)
-	r.Log.V(0).Info("Blueprint entrypoint: " + blueprintSpec.Entrypoint)
+	blueprintPerClusterMap := make(map[string]app.BlueprintSpec)
+	blueprintPerClusterMap[applicationContext.ClusterName] = *blueprintSpec
 
-	blueprint := r.GetBlueprintSignature(applicationContext)
```

Comment:
Am I correct in assuming that while you created the necessary structures, you have not yet implemented the code that decides which modules run in which cluster?

Reply:
Correct.